### ISSUE TITLE
FID score on CIFAR-10

### ISSUE THREAD

**Issue by JuntaoLiu01** [lqiang67/rectified-flow#29]
how do you infer & cal the fid score on CIFAR-10?

**Comment by CyberDragon93** [comment]
Hi, @JuntaoLiu01. We use the testing pipeline provided by [Clean FID](https://github.com/GaParmar/clean-fid), where we sample 50,000 images of rf and compute the FID score based on these samples.

**Comment by JuntaoLiu01** [comment]
Get & Thanks!

**Comment by JuntaoLiu01** [comment]
So could you provide the pretrained Unet or DiT model on CIFAR-10? the model weigths link in README.md seems to be unvalid and redirects to the url of the project?

**Comment by JuntaoLiu01** [comment]
Following the training scripts provided in README, I train the UNet model for 800K iterations (batch_size=128, GPU=8, steps=100K). The training loss converges at 0.125.
However, when I infer with the model, using the codes as bellows, the results don't satisfy.

```
def main(args):
    if args.seed is not None:
        set_seed(args.seed)

    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
    if args.allow_tf32 and torch.cuda.is_available():
        torch.backends.cuda.matmul.allow_tf32 = True

    UNet_config = SongUNetConfig(
        img_resolution=args.resolution,
        in_channels=3,
        out_channels=3,
    )
    model = SongUNet(UNet_config)
    model.from_pretrained(args.model_dir, filename="unet", use_ema=args.use_ema)
    model.eval()
    model = model.cuda()

    rectified_flow = RectifiedFlow(
        data_shape=(3, args.resolution, args.resolution),
        interp=args.interp,
        source_distribution=args.source_distribution,
        is_independent_coupling=args.is_independent_coupling,
        train_time_distribution=args.train_time_distribution,
        train_time_weight=args.train_time_weight,
        criterion=args.criterion,
        velocity_field=model,
        device=device,
    )

    num_sampling_steps = 100
    sampler = EulerSampler(
        rectified_flow=rectified_flow,
        num_steps=num_sampling_steps,
        num_samples=1,
    )

    # res = sampler.sample_loop_simple(num_samples=1, num_steps=num_sampling_steps)
    sampler.sample_loop(num_samples=1, num_steps=num_sampling_steps)
    res = sampler.trajectories[-1]
    res = res.permute(0, 2, 3, 1).detach().cpu().numpy()
    # print(np.min(res), np.max(res))
    out = np.clip(res * 127.5 + 128., 0, 255).astype(np.uint8)
    for idx in range(out.shape[0]):
        Image.fromarray(out[idx]).save(args.save_dir + "/%d.jpg" % idx)
        
    # print(res.size())
```

Can you help to give some advice?

**Comment by JuntaoLiu01** [comment]
Load model weights with a wrong method. The following codes are right:

```
UNet_config = SongUNetConfig(
        img_resolution=args.resolution,
        in_channels=3,
        out_channels=3,
    )
    model = SongUNet(UNet_config)

    filename = "unet"
    if args.use_ema:
        model_path = os.path.join(args.model_dir, f"{filename}_ema.pt")
    else:
        model_path = os.path.join(args.model_dir, f"{filename}_model.pt")
    state_dict = torch.load(model_path, map_location="cpu")
    model.load_state_dict(state_dict)
    # model = model.from_pretrained(args.model_dir, filename="unet", use_ema=args.use_ema)
    model.eval()
    model = model.cuda()
    print("model loaded done.")
```

**Comment by CyberDragon93** [comment]
> Following the training scripts provided in README, I train the UNet model for 800K iterations (batch_size=128, GPU=8, steps=100K). The training loss converges at 0.125. However, when I infer with the model, using the codes as bellows, the results don't satisfy.
> 
> ```
> def main(args):
>     if args.seed is not None:
>         set_seed(args.seed)
> 
>     device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
>     if args.allow_tf32 and torch.cuda.is_available():
>         torch.backends.cuda.matmul.allow_tf32 = True
> 
>     UNet_config = SongUNetConfig(
>         img_resolution=args.resolution,
>         in_channels=3,
>         out_channels=3,
>     )
>     model = SongUNet(UNet_config)
>     model.from_pretrained(args.model_dir, filename="unet", use_ema=args.use_ema)
>     model.eval()
>     model = model.cuda()
> 
>     rectified_flow = RectifiedFlow(
>         data_shape=(3, args.resolution, args.resolution),
>         interp=args.interp,
>         source_distribution=args.source_distribution,
>         is_independent_coupling=args.is_independent_coupling,
>         train_time_distribution=args.train_time_distribution,
>         train_time_weight=args.train_time_weight,
>         criterion=args.criterion,
>         velocity_field=model,
>         device=device,
>     )
> 
>     num_sampling_steps = 100
>     sampler = EulerSampler(
>         rectified_flow=rectified_flow,
>         num_steps=num_sampling_steps,
>         num_samples=1,
>     )
> 
>     # res = sampler.sample_loop_simple(num_samples=1, num_steps=num_sampling_steps)
>     sampler.sample_loop(num_samples=1, num_steps=num_sampling_steps)
>     res = sampler.trajectories[-1]
>     res = res.permute(0, 2, 3, 1).detach().cpu().numpy()
>     # print(np.min(res), np.max(res))
>     out = np.clip(res * 127.5 + 128., 0, 255).astype(np.uint8)
>     for idx in range(out.shape[0]):
>         Image.fromarray(out[idx]).save(args.save_dir + "/%d.jpg" % idx)
>         
>     # print(res.size())
> ```
> 
> Can you help to give some advice?

Hi, you should specify the folder which ckpts are in, rather than the path of the ckpt itself

**Comment by CyberDragon93** [comment]
> So could you provide the pretrained Unet or DiT model on CIFAR-10? the model weigths link in README.md seems to be unvalid and redirects to the url of the project?

Yes, the ckpt of UNet on CIFAR-10 https://drive.google.com/file/d/1_LeFSa25iYKGfsW7sbRFEabzEjRgDqYy/view?usp=sharing

**Comment by CyberDragon93** [comment]
Hi @JuntaoLiu01, we uploaded a new version of UNet on CIFAR-10, which ahieves FID of 2.5 with our training recipe
